command/apply.go

```diff
@@ -144,103 +206,75 @@ func (c *ApplyCommand) Run(args []string) int {
 				"Failed to read plan from plan file",
 				"The given plan file does not have a valid backend configuration. This is a bug in the Terraform command that generated this plan file.",
 			))
-			c.showDiagnostics(diags)
-			return 1
+			return nil, diags
 		}
 		be, beDiags = c.BackendForPlan(plan.Backend)
 	}
+
 	diags = diags.Append(beDiags)
 	if beDiags.HasErrors() {
-		c.showDiagnostics(diags)
-		return 1
+		return nil, diags
 	}
+	return be, diags
+}
+
+func (c *ApplyCommand) OperationRequest(be backend.Enhanced, view views.Apply, planFile *planfile.Reader, args *arguments.Operation,
+) (*backend.Operation, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
 
 	// Applying changes with dev overrides in effect could make it impossible
 	// to switch back to a release version if the schema isn't compatible,
 	// so we'll warn about it.
 	diags = diags.Append(c.providerDevOverrideRuntimeWarnings())
 
-	// Before we delegate to the backend, we'll print any warning diagnostics
-	// we've accumulated here, since the backend will start fresh with its own
-	// diagnostics.
-	c.showDiagnostics(diags)
-	diags = nil
-
 	// Build the operation
 	opReq := c.Operation(be)
-	opReq.AutoApprove = autoApprove
-	opReq.ConfigDir = configPath
+	opReq.AutoApprove = args.AutoApprove
+	opReq.ConfigDir = "."
 	opReq.Destroy = c.Destroy
-	opReq.Hooks = []terraform.Hook{countHook, c.uiHook()}
+	opReq.Hooks = view.Hooks()
 	opReq.PlanFile = planFile
-	opReq.PlanRefresh = refresh
-	opReq.ShowDiagnostics = c.showDiagnostics
+	opReq.PlanRefresh = args.Refresh
+	opReq.Targets = args.Targets
 	opReq.Type = backend.OperationTypeApply
-	opReq.View = views.NewOperation(arguments.ViewHuman, c.RunningInAutomation, c.View)
-
-	opReq.ConfigLoader, err = c.initConfigLoader()
-	if err != nil {
-		c.showDiagnostics(err)
-		return 1
+	opReq.View = view.Operation()
+	// FIXME: this shim is needed until the remote backend is migrated to views
+	opReq.ShowDiagnostics = func(vals ...interface{}) {
```

Comment:
To allow `error`s to be easily rendered, the `showDiagnostics` method accepts `...interface{}`. The local backend no longer needs this, as only `tfdiags.Diagnostics` values are used, but the remote backend still expects to be able to render errors for now. Once that is migrated, the backends can call `View.Diagnostics()` directly instead, even before all commands are fully using views: the global view has an implementation that works.

Reply:
Are you planning on making this change soon? What you wrote here in GH would be an excellent addition to the FIXME comment.

Reply:
Yes! It's on my list and it should be fairly simple to do. I'll update the FIXME 👍